<launch>
    <arg name="config_path" default = "$(find feature_tracker_)/../config/mynteye_leishen_outdoor.yaml" />
	<arg name="vils_path" default = "$(find feature_tracker_)/../output/" />

    <node name="feature_tracker_" pkg="feature_tracker_" type="feature_tracker_" output="log">
        <param name="config_file" type="string" value="$(arg config_path)" />
        <param name="vils_folder" type="string" value="$(arg vils_path)" />
    </node>

     <node name="lidar_compensator" pkg="lidar_compensator" type="lidar_compensator" output="log">
        <param name="lidar_topic" type="string" value="/lslidar_point_cloud"/>
       	<param name="rad_diff" type="double" value="1.0" />
  	    <param name="infer_start_ori" type="bool" value="false" />
        <param name="scan_period" type="double" value="0.1" />
    </node>

    <node name="vils_estimator" pkg="vils_estimator" type="vils_estimator" output="log">
       <param name="config_file" type="string" value="$(arg config_path)" />
       <param name="vils_folder" type="string" value="$(arg vils_path)" />
    </node>

    <param name="sensor_type" type="int" value="16" />
    <param name="minimum_range" type="double" value="0.1"/>

    <param name="mapping_line_resolution" type="double" value="0.2"/> 
    <param name="mapping_plane_resolution" type="double" value="0.4"/>

    <param name="save_directory" type="string" value="$(find feature_tracker_)/../output/"/> 
    <param name="performSC" type="int" value="1"/>
    <param name="save_map" type="int" value="0"/>
    <param name="onboard" type="int" value="1"/>
    <param name="sc_dist_thres" type="double" value="0.4"/>
    <param name="sc_max_radius" type="double" value="80.0"/>
    <!-- set >20 for larger scale outdoor;  set 10~20 for outdoor(altitude shift) -->
    <param name="floorHeight" type="double" value="12.0"/>  
       
    <node pkg="vils_mapping" type="feature" name="feature" output="log" />
    <node pkg="vils_mapping" type="local" name="local" output="log" />
    <node pkg="vils_mapping" type="global_octree" name="global_octree" output="log" />
    <!-- <node pkg="vils_mapping" type="global_ikdtree" name="global_ikdtree" output="screen" /> -->

    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find vils_estimator)/../config/vils_rviz_config.rviz" />
    </group>

</launch>
